sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala

```diff
@@ -406,6 +408,33 @@ class SparkSessionExtensionSuite extends SparkFunSuite {
       session.sql("SELECT * FROM v")
     }
   }
+
+  test("SPARK-38697: Extend SparkSessionExtensions to inject rules into AQE Optimizer") {
+    def checkLimit(df: Dataset[java.lang.Long], hasLimit: Boolean): Unit = {
+      def executedPlan: SparkPlan = df.queryExecution.executedPlan match {
+        case aqe: AdaptiveSparkPlanExec => aqe.executedPlan
+        case other => other
```

Comment:
let's explicitly turn on AQE in the test, so that we don't need this `other` case. This feature is available in AQE anyway.

Reply:
assert here